Histfit doc (#258)
PhiLFitters/kafe2: doc/src/parts/user_guide.rst

    called with the corresponding keyword :code:`fit.do_fit(asymmetric_parameter_errors=True)`.
    Otherwise they will be :py:obj:`None`.

Histogram Fits
---------------

> A very common fit type is the histogram fit. *kafe2* provides a dedicated fitting class for histogram fits,

Reviewer: ```suggestion
In physics experiments data is frequently histogrammed in order to reduce the data to a manageable number of bins.
*kafe2* provides a dedicated fitting class for histogram fits,
```

> *kafe2* provides a dedicated fitting class for histogram fits, intendet to be used when datapoints are obtained

Reviewer: ```suggestion
*kafe2* provides a dedicated fitting class for histogram fits, intended to be used when datapoints are obtained
```

> from a random distribution. Especiallywhen large numbers of datapoints are present it is more efficient to treat

Reviewer: ```suggestion
from a random distribution. Especially when large numbers of datapoints are present it is more efficient to treat
```

> the data as a histogram. To perform a histogram fit, the datapoints have to be filled into a :py:obj:`HistContainer`.

Reviewer: ```suggestion
the data as a histogram. To perform a histogram fit using raw data, the datapoints have to be filled into a :py:obj:`HistContainer`.
```

is similar to the above. By default, the :py:obj:`HistFit` class will use a normal distribution as model
function and a poisson likelihood as cost function. Both can be changed using the `model_function` and 
`cost_function` keywords.

.. code-block:: python

    from kafe2 import Fit, HistContainer

    histogram = HistContainer(n_bins=10, bin_range=(-5, 5),
                          fill_data=[-7.5, 1.23, 5.74, 1.9, -0.2, 3.1, -2.75, ...])
    hist_fit = Fit(histogram)
    hist_fit.do_fit()
    
Depending on whether the modelfunction is already normalised, or has a normalisation constant,
that is also supposed to be estimated in the fit the `density` keyword can be used, during creation

> of the `HistFit` object.

Reviewer: ```suggestion
By default it is assumed that the model function for a `HistFit` object is a probability density function normalised to 1.
As a consequence the bin contents are also being normalized to 1.
To disable this behavior, set `density=False` in the `HistFit` constructor.
```

I think this explains the behavior more explicitly. We can still mention the fitting of normalization constants, though you could in principle also have a model function with a constant normalization that you're not fitting.

> As a consequence the bin contents are also being normalized to 1.

Reviewer: ```suggestion
By default it is assumed that the model function for a `HistFit` object is a probability density function normalized to 1.
As a consequence the bin contents are also being normalized to 1.
```

I don't particularly care which spelling convention we use but we should be consistent.
